unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/TaskbarHUD/TaskbarHUDController.cs

```diff
@@ -112,177 +90,255 @@ public struct Configuration
 
         numOfLoadedExperiences.OnChange += NumOfLoadedExperiencesChanged;
         NumOfLoadedExperiencesChanged(numOfLoadedExperiences.Get(), 0);
-    }
 
-    private void ChatHeadsGroup_OnHeadClose(TaskbarButton obj) { privateChatWindowHud.SetVisibility(false); }
+        if (friendsController != null)
+        {
+            if (friendsController.isInitialized)
+            {
+                view.SetFiendsAsLoading(false);
+            }
+            else
+            {
+                friendsController.OnInitialized -= HandleFriendsInitialization;
+                friendsController.OnInitialized += HandleFriendsInitialization;
+                view.SetFiendsAsLoading(true);
+            }
+        }
+    }
 
-    private void View_OnFriendsToggleOn()
+    private void HandleFriendsToggle(bool show)
     {
-        friendsHud?.SetVisibility(true);
+        if (show)
+            OpenFriendsWindow();
+        else
+            friendsHud.SetVisibility(false);
         OnAnyTaskbarButtonClicked?.Invoke();
     }
 
-    private void View_OnFriendsToggleOff() { friendsHud?.SetVisibility(false); }
+    private void HandleEmotesToggle(bool show)
+    {
+        if (show)
+            ShowEmotes();
+        else
+            isEmotesVisible.Set(false);
+        OnAnyTaskbarButtonClicked?.Invoke();
+    }
 
-    private void View_OnEmotesToggleOn()
+    private void ShowEmotes()
     {
+        worldChatWindowHud.SetVisibility(false);
+        privateChatWindow.SetVisibility(false);
+        publicChatChannel.SetVisibility(false);
+        friendsHud.SetVisibility(false);
+        isExperiencesViewerOpen.Set(false);
         isEmotesVisible.Set(true);
-        OnAnyTaskbarButtonClicked?.Invoke();
+        view.ToggleOn(TaskbarHUDView.TaskbarButtonType.Emotes);
     }
 
-    private void View_OnEmotesToggleOff() { isEmotesVisible.Set(false); }
-    
-    private void View_OnExperiencesToggleOn()
+    private void HandleExperiencesToggle(bool show)
     {
-        isExperiencesViewerOpen.Set(true);
+        if (show)
+            ShowExperiences();
+        else
+            isExperiencesViewerOpen.Set(false);
         OnAnyTaskbarButtonClicked?.Invoke();
     }
 
-    private void View_OnExperiencesToggleOff() { isExperiencesViewerOpen.Set(false); }
+    private void ShowExperiences()
+    {
+        worldChatWindowHud.SetVisibility(false);
+        privateChatWindow.SetVisibility(false);
+        publicChatChannel.SetVisibility(false);
+        friendsHud.SetVisibility(false);
+        isEmotesVisible.Set(false);
+        isExperiencesViewerOpen.Set(true);
+    }
 
     private void ToggleFriendsTrigger_OnTriggered(DCLAction_Trigger action)
     {
-        if (!view.friendsButton.transform.parent.gameObject.activeSelf)
+        if (!friendsController.isInitialized)
             return;
 
-        OnFriendsToggleInputPress();
+        // ??????
```

Comment:
What that comment means? If you think we should remark something here, could we change it by something more descriptive?